tests/main/snapshot-exclusions-dynamic/task.yaml

```diff
@@ -0,0 +1,124 @@
+    SET_ID=$( echo "$RESPONSE" | edge-jq '.result."set-id"' )
+    echo "$SET_ID" | MATCH "^[0-9]+$"
+    CHANGE=$( echo "$RESPONSE" | edge-jq ".change" | grep -o "[0-9]*" )
+    echo "$CHANGE" | MATCH "^[0-9]+$"
+
+    # Wait for completion of async change
+    retry -n 20 --wait 0.1 sh -c "edge-curl -s --max-time 5 --unix-socket /run/snapd.socket http://localhost/v2/changes/$CHANGE | edge-jq '.result.tasks[0].progress.done' | MATCH 1"
+
+    # Confirm the snapshot for the set ID includes the test-snap
+    snap saved --id="$SET_ID" | MATCH "test-snap"
+    # and is valid
+    snap check-snapshot "$SET_ID"
+    # and the snapshot metadata includes options
+    edge-curl -s --unix-socket /run/snapd.socket --max-time 5 http://localhost/v2/snapshots?set="$SET_ID" | MATCH '"options":'
+
+    # Check that shapshot is exportable
+    snap export-snapshot "$SET_ID" "${SET_ID}_export.snapshot"
+    test -e "${SET_ID}_export.snapshot"
+    # and importable
+    snap import-snapshot "${SET_ID}_export.snapshot"
+    # and the snapshot metadata includes the expected options entry (dynamic exclusions only)
+    cat <<EOF > expected_options_entry
+    "\$SNAP_DATA/dynamic-exclude.txt"
+    "\$SNAP_COMMON/dynamic-exclude.txt"
+    "\$SNAP_USER_COMMON/dynamic-exclude.txt"
+    "\$SNAP_USER_DATA/dynamic-exclude.txt"
+    EOF
+    edge-curl -s --unix-socket /run/snapd.socket --max-time 5 http://localhost/v2/snapshots?set="$SET_ID" | edge-jq .result[0].snapshots[0].options.exclude[] > actual_options_entry
+    diff -u expected_options_entry actual_options_entry
+
+    # Remove the canaries to test restore
+    for DIR in "$USER_DATA_DIR" "$SYSTEM_DATA_DIR"
+    do
+        rm "$DIR/"{current,common}/canary.txt
+    done
+    
+    # Check restore
+    snap restore "$SET_ID" test-snap
+    MATCH "system versioned" < "$SYSTEM_DATA_DIR/current/canary.txt"
+    MATCH "system common" < "$SYSTEM_DATA_DIR/common/canary.txt"
+    MATCH "user versioned" < "$USER_DATA_DIR/current/canary.txt"
+    MATCH "user common" < "$USER_DATA_DIR/common/canary.txt"
+    cat <<EOF > expected_restored_files
+    $USER_DATA_DIR
+    $USER_DATA_DIR/common
+    $USER_DATA_DIR/common/canary.txt
+    $USER_DATA_DIR/current
+    $USER_DATA_DIR/x1
+    $USER_DATA_DIR/x1/canary.txt
+    $SYSTEM_DATA_DIR
+    $SYSTEM_DATA_DIR/common
+    $SYSTEM_DATA_DIR/common/canary.txt
+    $SYSTEM_DATA_DIR/current
+    $SYSTEM_DATA_DIR/x1
+    $SYSTEM_DATA_DIR/x1/canary.txt
+    EOF
+    find "$USER_DATA_DIR" "$SYSTEM_DATA_DIR" | sort | grep -v '\.~\w*~' > actual_restored_files
+    diff -u expected_restored_files actual_restored_files
+
+    # Test permutations of incomplete dynamic snapshot options that is valid JSON
```

Comment:
IMO, these checks might be a bit too detailed for a spread test. We already have them as unit tests where I think they're more appropriate. The spread test's purpose is more to check the feature functioning with the whole system so we don't need to check every possible input

Reply:
I was on the fence on these myself. Thanks, removed.